vendor/engines/ldap_authentication/lib/ldap_authentication/user_converter.rb

```diff
@@ -0,0 +1,23 @@
+module LdapAuthentication
+
+  class UserConverter
+
+    CONVERTABLE_ATTRIBUTES = [:username, :first_name, :last_name, :email].freeze
+
+    def initialize(user_entry)
+      @user_entry = user_entry
+    end
+
+    def to_user
+      ::User.new(attributes)
+    end
+
+    def attributes
+      CONVERTABLE_ATTRIBUTES.each_with_object({}) do |field, output|
+        output[field] = @user_entry.public_send(field)
+      end
+    end
```

Comment:
I started to play around with a `UserEntry#to_h`, but then `UserConverter` became almost nothing, and it felt weird to move the `::User` information into `UserEntry`. I prefer `UserEntry` just being the LDAP information and then the `UserConverter` to know about which fields from LDAP it can bring over into `::User`.